File: izlaganja/predavanja/2007-08/prezentacija-20_prilozi/PHP/udp_server.php
<?php

	//provjera broja argumenata skripte
	if( $argc != 2 ) {
		echo "usage: php udp_server.php <port>\n";
		echo "port between 1024 and 65535\n";
		exit(1);
	}

	$socket = @socket_create(AF_INET, SOCK_DGRAM, SOL_UDP);
	if( $socket == FALSE ) reportError();
	if( socket_bind($socket, '127.0.0.1', $argv[1]) == FALSE ) reportError();
	echo "Ready to accept connections at port $argv[1]\n";

	do {
		$c_addr = "";
		$c_port = 0;
		
		$len = @socket_recvfrom($socket, $msg, 1500, 0, $c_addr, $c_port);
		if($len == -1) reportError();
		else if($len == 0) continue;
		
		echo "Query received from " . $c_addr . " at port " . $c_port . ": $msg\n";
		
		if( strcasecmp($msg, "time") == 0 ) $rsp = date("D M j H:i:s Y\r\n");
		else $rsp = date("not understood");
			
		if( socket_sendto($socket, $rsp, strlen($rsp), 0, $c_addr, $c_port ) == -1 )
			reportError();
		
	} while( TRUE );

	socket_close($socket);

	function reportError() {
	
		echo "(" . socket_last_error() . ") " . socket_strerror(socket_last_error()). "\n";
		exit(1);
	}
?>

File: izlaganja/predavanja/2007-08/prezentacija-20_prilozi/Java/udp_server.java
import java.io.IOException;
import java.net.DatagramPacket;
import java.net.DatagramSocket;
import java.net.SocketException;
import java.util.Date;


public class udp_server {

	public static void main(String[] args) {
		
		if( args.length != 1 ) {
			System.out.println("usage: java udp_server <port>");
			System.out.println("port between 1024 and 65535");
			System.exit(1);
		}
		
		DatagramSocket ssocket = null;
		
		try {
			ssocket = new DatagramSocket( Integer.parseInt(args[0]));
			System.out.println("Ready to accept connections at port " + args[0]);
		} catch (NumberFormatException e) {
			System.out.println("Argument is not a valid port number: " + args[0]);
			System.exit(1);
		} catch (SocketException e) {
			System.out.println("Error creating a server datagram socket at port " + args[0]);
			System.exit(1);
		}
		
		while(true) {
			
			byte[] buffer = new byte[100];
			DatagramPacket dPacket = new DatagramPacket(buffer, buffer.length);
			String message = null;
			try {
				ssocket.receive(dPacket);
				message = new String(buffer);
				System.out.println("Query received from " + dPacket.getAddress() + " at port " + dPacket.getPort() + ": " + message);
			} catch (IOException e) {
				System.out.println("Error receiving a datagram packet." );
				continue;
			}
			
			String response = null;
			if( message.trim().equalsIgnoreCase("time") )
				response = new Date().toString();
			else
				response = "not understood";
			
			byte[] rBuffer = response.getBytes();
			
			DatagramPacket rPacket = new DatagramPacket(rBuffer, rBuffer.length, dPacket.getAddress(), dPacket.getPort());
			try {
				ssocket.send(rPacket);
			} catch (IOException e) {
				System.out.println("Error sending a datagram packet." );
				System.exit(1);
			}
			
		}
	}
}
